internal/service/alarm_notification_handler.go

```diff
@@ -0,0 +1,287 @@
+	b.kubeClient = kubeClient
+	return b
+}
+
+// Build uses the data stored in the builder to create anad configure a new handler.
+func (b *alarmNotificationHandlerBuilder) Build(ctx context.Context) (
+	result *alarmNotificationHandler, err error) {
+	// Check parameters:
+	if b.logger == nil {
+		err = errors.New("logger is mandatory")
+		return
+	}
+	if b.cloudID == "" {
+		err = errors.New("cloud identifier is mandatory")
+		return
+	}
+
+	if b.kubeClient == nil {
+		err = errors.New("kubeClient is mandatory")
+		return
+	}
+
+	// Prepare the JSON iterator API:
+	jsonConfig := jsoniter.Config{
+		IndentionStep: 2,
+	}
+	jsonAPI := jsonConfig.Froze()
+
+	// Create the filter expression evaluator:
+	pathEvaluator, err := search.NewPathEvaluator().
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+	selectorEvaluator, err := search.NewSelectorEvaluator().
+		SetLogger(b.logger).
+		SetPathEvaluator(pathEvaluator.Evaluate).
+		Build()
+	if err != nil {
+		return
+	}
+	// Create the jq tool:
+	jqTool, err := jq.NewTool().
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+
+	// Check that extensions are at least syntactically valid:
+	for _, extension := range b.extensions {
+		_, err = jqTool.Compile(extension)
+		if err != nil {
+			return
+		}
+	}
+
+	alarmSubscriptionSearcher := newAlarmSubscriptionSearcher()
+	alarmSubscriptionSearcher.SetLogger(b.logger).SetJqTool(jqTool).build()
+
+	// create persist storeage option
+	persistStore := persiststorage.NewKubeConfigMapStore().
+		SetNameSpace(TestNamespace).
+		SetName(TestConfigmapName).
+		SetFieldOwnder(FieldOwner).
+		SetJsonAPI(&jsonAPI).
+		SetClient(b.kubeClient)
+
+	// http client to send out notification
+	// use 2 sec first
+	httpClient := http.Client{Timeout: 2 * time.Second}
+
+	if singleAlarmNotificationHandle != nil {
+		panic("There is an existing handler instance")
+	}
+	// Create and populate the object:
+	result = &alarmNotificationHandler{
+		logger:                    b.logger,
+		loggingWrapper:            b.loggingWrapper,
+		cloudID:                   b.cloudID,
+		extensions:                slices.Clone(b.extensions),
+		selectorEvaluator:         selectorEvaluator,
+		jsonAPI:                   jsonAPI,
+		jqTool:                    jqTool,
+		subscriptionMapMemoryLock: &sync.RWMutex{},
+		subscriptionMap:           &map[string]data.Object{},
+		persistStore:              persistStore,
+		subscriptionSearcher:      alarmSubscriptionSearcher,
+		httpClient:                httpClient,
+	}
+	if singleAlarmNotificationHandle != nil {
+		panic("There is an existing handler instance")
+	}
+	singleAlarmNotificationHandle = result
+
+	b.logger.Debug(
+		"alarmNotificationHandler build:",
+		"CloudID", b.cloudID,
+	)
+
+	err = result.recoveryFromPersistStore(ctx)
+	if err != nil {
+		b.logger.Error(
+			"alarmNotificationHandler failed to recovery from persistStore ",
+			slog.String(": ", err.Error()),
```

Comment:
This will return in JSON log message like this:

```json
{
  ": ": "the details of the error"
}
```

That looks weird. Can we use `slog.String("error", err.Error())` instead?